### ISSUE TITLE
[Enhancement] Add smooth scrolling effect to tab bar

### ISSUE THREAD

**Issue by kyaso** [Zettlr/Zettlr#3243]
### Description

I played a bit with the tab bar, and found that adding `scroll-behavior: smooth;` to the tabs div, gives the user a nice visual feedback when switching to a tab that is out of view.

Besides being a UI candy, the scroll animation also serves the purpose of indicating _where_ (to be more precise: in which direction) the target tab is located relative to the origin tab.

What are your thoughts?

### Proposed Changes

Add `scroll-behavior: smooth;` somewhere in this section:

https://github.com/Zettlr/Zettlr/blob/afa6ddfaf325e3e8219219a690ce62b6a65726f6/source/win-main/DocumentTabs.vue#L399-L409

### Caveats

Two potential issues:
* Are those kinds of animations welcome in Zettlr?
* When you have _many_ tabs open, and you switch between the first and the last tab, the scrolling effect could be annoying to some people. The file you switched to is opened instantly, while the scroll animation is only finished after one or two seconds.
    * But I believe that this scenario is rare
    * On the other hand, this could also be seen as a hint: _"Hey, you might have two many tabs open, maybe do some clean up?"_
        * Currently, with the static switching, you can't really tell that you have that many tabs open

### Do you Wish to Attempt Implementing this Yourself?

Yes

### Zettlr Version

Stable (most recent version)

### Your Platform

- [X] Windows
- [X] macOS
- [X] Linux

**Comment by nathanlesage** [comment]
Fascinating what kinds of CSS properties we now have available.

In general sounds like a nice idea. Here are some thoughts:

* Animations are always welcome as long as they …
	* … are quick enough so you don't have to wait for them to finish (iOS, I'm looking at you! 😠 )
	* … are non-intrusive
	* … do not seem weird with respect to how "real" native applications do it

That being said: If there is an official CSS property for that, chances are high that this is an effect that is expected on many platforms and it should therefore be perfectly fine to implement it. Another way to check for this is to look at how native applications with tabbed interfaces handle this. On macOS I know that tabbed windows do expose precisely this behavior.

Just tried that out and it looks quite cool. However, what worries me is that this effect seems somewhat slow – you mention 1-2 seconds, and I also saw that the effect seems a little bit slow. The scrolling is not really limited to a _maximum_ number, which kind of worries me.

But then, this is an effect that everyone could super easily disable with custom CSS, so the friction for people who like the snappy behavior would be very low (as long as we announce that change in behavior :D)

**Comment by kyaso** [comment]
> But then, this is an effect that everyone could super easily disable with custom CSS, so the friction for people who like the snappy behavior would be very low (as long as we announce that change in behavior :D)

I've created a PR! Plus, I also included your custom CSS hint in the changelog (if that's the right place for it?) :D

**Comment by nathanlesage** [comment]
> in the changelog (if that's the right place for it?) :D

Yes! People either read that stuff, or they don't deserve better, hehe
